lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -379,6 +386,9 @@ def validate_tags(self, tags):
             for key, value in tags.items():
                 if not isinstance(value, str):
                     self.fail("Tags values must be strings. Found {0}:{1}".format(str(key), str(value)))
+        else:
+            if not isinstance(tags, list) and not isinstance(tags, dict):
+                self.fail("Tags must be a dictionary of string:string values or list of string values in info modules")
```

Comment:
does it need to check value when the tag is dict?

Reply:
From my view, the ideal solution is to make sure that ```tags``` is the list type in the facts module. However, there are already many facts modules and I'm not sure that all of them supports the list type. A soft check(supports list and dict) may avoid possible errors. For the new module, if we use ```AZURE_INFO_TAG_ARGS```, the type of tags will be limited to list.